### PR TITLE
Add protected middleware handler

### PR DESCRIPTION
**Pull request by shan-96**
Can create an issue for this

- This is extension for Basic auth handler since it gives more options

### CODE REVIEW

**Comment by shan-96** [comment]
@ksysoev have a look at this whenever you can and let me know your thoughts

File: middleware/http/utils.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package http
+
+import "net/http"
+
+// Unauthorized sends an HTTP 401 Unauthorized response with the specified operations made on responseWriter.
+func Unauthorized(w http.ResponseWriter, message string, writerPostOperations ...func(w http.ResponseWriter)) {
```
Comment (ksysoev) [inline middleware/http/utils.go:6]: we can keep this private, we can use private functions within the same package.

File: middleware/http/protected.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+				return
+			}
+
+			tokenString = tokenString[len("Bearer "):]
+
+			err := verifyToken(tokenString)
+			if err != nil {
+				Unauthorized(w, "Invalid token", func(w http.ResponseWriter) { w.WriteHeader(http.StatusUnauthorized) })
+				return
+			}
```
Comment (ksysoev) [inline middleware/http/protected.go:27]: `Unauthorized` already setting `http.StatusUnauthorized`, we  probably can drop writerPostOperations here

**Review by ksysoev — APPROVED** [review]
LGTM, @shan-96 thank you for the contribution

**Pull request merged**